Formula/libtasn1.rb

```diff
@@ -20,6 +20,8 @@ class Libtasn1 < Formula
   end
 
   def install
+    # https://gitlab.com/gnutls/libtasn1/-/issues/30
+    ENV.O1 if DevelopmentTools.clang_build_version >= 1200
```

Comment:
`ENV#O1` [has been disabled](https://github.com/Homebrew/brew/blob/04dab8428b4db1bb94bc1284cbaf77322fef9334/Library/Homebrew/extend/ENV/super.rb#L347-L349), so we shouldn't be using this.

Reply:
@carlocab what should be used instead?

Reply:
Passing the `-O1` flag through environment variables probably won't work because of environment filtering... So I guess our options here are to either:
1. use the patch at https://gitlab.com/gnutls/libtasn1/-/merge_requests/71; or
2. compile with `gcc` instead (use `depends_on "gcc"` and `fails_with: :clang`, [ref](https://docs.brew.sh/Formula-Cookbook#compiler-selection))

Thoughts, @Homebrew/core?

Reply:
Yep, `test-bot` won't let you use it:
```
Error: An exception occurred within a child process:
  MethodDeprecatedError: Calling ENV.O1 is disabled! There is no replacement.
```

Reply:
> 2\. compile with `gcc` instead (use `depends_on "gcc"` and `fails_with :clang`, [ref](https://docs.brew.sh/Formula-Cookbook#compiler-selection))

If this compile C++ code this will introduce a runtime dependency, just FYI.



> * use the patch at https://gitlab.com/gnutls/libtasn1/-/merge_requests/71; or

This seems preferable 👍🏻

Reply:
I thought `libtasn1` was pure C, so I tried to rebuild it locally with `gcc`. However, it creates a runtime dependency:
```
❯ brew linkage libtasn1
System libraries:
  /usr/lib/libSystem.B.dylib
Homebrew libraries:
  /usr/local/lib/gcc/10/libgcc_s.1.dylib (gcc)
  /usr/local/Cellar/libtasn1/4.16.0/lib/libtasn1.6.dylib (libtasn1)
Undeclared dependencies with linkage:
  gcc
```
So if the patch is good, we should use that. However, `libtasn1` is used in security-sensitive formulae (e.g. `gnutls`), so I think a runtime dependency on `gcc` would be safer, if there's any uncertainty at all regarding the quality of the patch.

Reply:
@carlocab @MikeMcQuaid thanks for your comments. I think it's worth to wait a reply from libtasn1 maintainer before going further.